spring-integration-core/src/main/java/org/springframework/integration/support/SmartLifecycleRoleController.java

```diff
@@ -185,7 +190,76 @@ public void stopLifecyclesInRole(String role) {
 		}
 	}
 
-	private void addLazyLifecycles() {
+	/**
+	 * Return a collection of the roles currently managed by this controller.
+	 * @return the roles.
+	 * @since 4.3.8
+	 */
+	public Collection<String> getRoles() {
+		if (this.lazyLifecycles.size() > 0) {
+			addLazyLifecycles();
+		}
+		return new ArrayList<>(this.lifecycles.keySet());
+	}
+
+	/**
+	 * Return true if all endpoints in the role are running.
+	 * @param role the role.
+	 * @return true if at least one endpoint in the role, and all are running.
+	 * @since 4.3.8
+	 */
+	public boolean allEndpointsRunning(String role) {
+		Map<String, Boolean> status = getEndpointsRunningStatus(role);
+		if (status.isEmpty()) {
+			return false;
+		}
+		else {
+			return status.values().stream().allMatch(b -> b);
+		}
+	}
+
+	/**
+	 * Return true if none of the endpoints in the role are running or if
+	 * there are no endpoints in the role.
+	 * @param role the role.
+	 * @return true if there are no endpoints or none are running.
+	 * @since 4.3.8
+	 */
+	public boolean noEndpointsRunning(String role) {
+		Map<String, Boolean> status = getEndpointsRunningStatus(role);
+		if (status.isEmpty()) {
+			return true;
+		}
+		else {
+			return status.values().stream().allMatch(b -> !b);
```

Comment:
Negative ``allMatch` can be replaced with the `noneMatch`.
Plus `if...else` can be simplified too:
```
return status.isEmpty() || status.values().stream().noneMatch(b -> b);
```
That is what IDEA does for me! 😄